lib/sensors/src/tof.rs

```diff
@@ -0,0 +1,386 @@
+            SYS_INTERRUPT_CONFIG_GPIO,
+            SYS_INTERRUPT_CONFIG_GPIO_VAL,
+        ) {
+            return Err(ToFError::I2cError(e));
+        }
+
+        // Write 0x00 to SYS_FRESH_OUT_RESET to indicate that the sensor has been configured
+        // Note: as above, this means that the sensor will need to be power cycled or this register will need to be written to 0x01 to reconfigure the sensor
+        if let Err(e) =
+            i2c.write_byte_to_register_16(device_address, SYSTEM_FRESH_OUT_OF_RESET, 0x00)
+        {
+            return Err(ToFError::I2cError(e));
+        }
+        defmt::info!("ToF sensor configured");
+
+        Ok(Self {
+            i2c,
+            device_address,
+        })
+    }
+
+    pub fn start_ss_measure(&mut self) -> Result<i32, ToFError> {
+        // start single shot measurement
+        if let Err(e) = self.i2c.write_byte_to_register_16(
+            self.device_address,
+            SYSRANGE_START,
+            SYSRANGE_START_SS_VAL,
+        ) {
+            return Err(ToFError::I2cError(e));
+        }
+        Ok(1)
+    }
+
+    pub async fn poll_range(&mut self) {
+        let status_byte = self
+            .i2c
+            .read_byte_16(self.device_address, RESULT_INTERR_STATUS_GPIO)
+            .unwrap_or(0);
+        let mut range_status = status_byte & 0x07;
+        while range_status != 0x04 {
+            defmt::debug!("Range status: {:?}", range_status);
+            range_status = self
+                .i2c
+                .read_byte_16(self.device_address, RESULT_INTERR_STATUS_GPIO)
+                .unwrap_or_default()
+                & 0x07;
+            Timer::after(Duration::from_millis(100)).await;
+        }
+    }
+
+    pub fn read_range(&mut self) -> Option<u8> {
+        let range_byte = match self.i2c.read_byte_16(self.device_address, RESULT_RANGE_VAL) {
+            Some(byte) => byte,
+            None => {
+                return None;
+            }
+        };
+        Some(range_byte)
+    }
+
+    pub fn start_cts_measure(&mut self) -> Result<i32, ToFError> {
+        // start continuous measurement
+        if let Err(e) = self.i2c.write_byte_to_register_16(
+            self.device_address,
+            SYSRANGE_START,
+            SYSRANGE_START_CTS_VAL,
+        ) {
+            return Err(ToFError::I2cError(e));
+        }
+        Ok(1)
+    }
+
+    pub fn check_reset(&mut self) -> bool {
+        let reset_value = self
+            .i2c
+            .read_byte_16(self.device_address, SYS_FRESH_OUT_RESET)
+            .unwrap_or(0);
+        reset_value == 1
+    }
+
+    pub fn clear_interrupts(&mut self) -> Result<i32, ToFError> {
+        // at the end clear interrupts
+        if let Err(e) = self.i2c.write_byte_to_register_16(
+            self.device_address,
+            SYS_INTERRUPT_CLEAR,
+            CLEAR_INTERRUPTS_VAL,
+        ) {
+            return Err(ToFError::I2cError(e));
+        }
+        Ok(1)
+    }
```

Comment:
When will this be called?

Reply:
Clear Interrupts is called at the end of the reading cycle as seen on page 22 of the [application sheet](https://www.st.com/resource/en/application_note/an4545-vl6180x-basic-ranging-application-note-stmicroelectronics.pdf); I'll add it to the ToF task since it's missing from there